Signal/src/call/CallService.swift

```diff
@@ -109,33 +108,28 @@ protocol CallServiceObserver: class {
 
     static let fallbackIceServer = RTCIceServer(urlStrings: ["stun:stun1.l.google.com:19302"])
 
-    // Synchronize call signaling on the callSignalingQueue to make sure any appropriate requisite state is set.
-    static let signalingQueue = DispatchQueue(label: "CallServiceSignalingQueue")
-
     // MARK: Ivars
 
     var peerConnectionClient: PeerConnectionClient?
     // TODO code cleanup: move thread into SignalCall? Or refactor messageSender to take SignalRecipient identifier.
     var thread: TSContactThread?
     var call: SignalCall? {
         didSet {
-            assertOnSignalingQueue()
+            AssertIsOnMainThread()
 
             oldValue?.removeObserver(self)
             call?.addObserverAndSyncState(observer: self)
 
-            DispatchQueue.main.async { [weak self] in
-                self?.updateIsVideoEnabled()
-            }
```

Comment:
Most of this PR consists of this kind of simplification.  Most everything moves to the main thread.